libs/datasets/dataset_orchestrator.py

```diff
@@ -0,0 +1,318 @@
+        generate_cbsas: bool = True,
+        generate_hsas: bool = True,
+    ) -> MultiRegionDataset:
+        regions = [region for dataset in datasets for region in dataset.location_ids]
+        assert len(regions) == len(set(regions)), "Can't combine datasets with duplicate locations"
+
+        _log.info("Starting region combination...")
+        timeseries_df = (
+            pd.concat([dataset.timeseries_bucketed for dataset in datasets])
+            .sort_index()
+            .rename_axis(columns=PdFields.VARIABLE)
+        )
+        static_df = (
+            pd.concat(dataset.static for dataset in datasets)
+            .sort_index()
+            .rename_axis(columns=PdFields.VARIABLE)
+        )
+        tag = pd.concat([dataset.tag for dataset in datasets]).sort_index()
+        multiregion_dataset = MultiRegionDataset(
+            timeseries_bucketed=timeseries_df, static=static_df, tag=tag
+        )
+
+        if generate_hsas:
+            _log.info("Starting HSA aggregation...")
+            hsa_aggregator = statistical_areas.CountyToHSAAggregator.from_local_data()
+            multiregion_dataset = hsa_aggregator.aggregate(multiregion_dataset)
+
+        if generate_cbsas:
+            _log.info("Starting CBSA aggregation...")
+            cbsa_dataset = self.cbsa_aggregator.aggregate(
+                multiregion_dataset, reporting_ratio_required_to_aggregate=DEFAULT_REPORTING_RATIO
+            )
+            multiregion_dataset = multiregion_dataset.append_regions(cbsa_dataset)
+            multiregion_dataset.print_stats("CBSA dataset")
+
+        if aggregate_to_country:
+            _log.info("Starting country aggregation...")
+            multiregion_dataset = custom_aggregations.aggregate_to_country(
+                multiregion_dataset, reporting_ratio_required_to_aggregate=DEFAULT_REPORTING_RATIO
+            )
+            multiregion_dataset.print_stats("Aggregate to country")
+        return multiregion_dataset
+
+    def _build_region_timeseries(self, region_dataset: OneStateDataset):
+        state = region_dataset.state
+        multiregion_dataset = region_dataset.multiregion_dataset
+
+        # TODO: maybe filter overrides to match only the current region?
+        # Not sure if that will help performance but worth a shot.
+        region_overrides_config = manual_filter.transform_region_overrides(
+            self.region_overrides, self.cbsa_aggregator.cbsa_to_counties_region_map
+        )
+        before_manual_filter = multiregion_dataset
+        multiregion_dataset = manual_filter.run(multiregion_dataset, region_overrides_config)
+        manual_filter_touched = manual_filter.touched_subset(
+            before_manual_filter, multiregion_dataset
+        )
+        manual_filter_touched.write_to_wide_dates_csv(
+            dataset_utils.MANUAL_FILTER_REMOVED_WIDE_DATES_CSV_PATH,
+            dataset_utils.MANUAL_FILTER_REMOVED_STATIC_CSV_PATH,
+        )
```

Comment:
Hrm...  I think this might end up overwriting these files for every state in parallel.  Since we don't use Tom's data dashboard anymore, we could potentially just get rid of these files.  Else I think the right solution is probably to store the touched records in the state_dataset somehow and then combine / write them to the CSV after all states have been processed.

Reply:
ooop thank you :( The easiest way I found to deal with this was to just return this filtered dataset alongside the other data, and combine them in `build_and_combine_regions` (see efa6f92d89d1b944afb592940ea1de51cd46b118)